canonical-web-and-design/snapcraft.io
# Drag'n'drop to promote channel

Fixes #1961 
Fixes #1962 
Fixes #1963 

Adds drag'n'drop to releases page.
Channels can now be promoted using drag'n'drop.

### QA

This is new quite complex feature. Please make sure to QA on different snaps (different number of architectures, devmode revisions, closed channels, etc) and in different browsers (it uses native drag'n'drop, so browser implementations may differ).

- ./run or https://snapcraft-io-canonical-web-and-design-pr-2019.run.demo.haus/
- go to releases page or any snap (many snaps ideally)
- try to promote some stuff using drag'n'drop
- make sure that things that can be normally promoted can also be promoted with drag'n'drop
- make sure that things that can't be normally promoted (empty channel, devmode) can't be promoted with drag'n'drop
- click on things, try drag'n'drop in different UI states

<img width="1162" alt="Screenshot 2019-06-25 at 16 35 31" src="https://user-images.githubusercontent.com/83575/60108593-20d98f80-9769-11e9-95c7-04223adc26d8.png">
